### ISSUE TITLE
Library is broken in environment using CommonJS modules

### ISSUE THREAD

**Issue by devoto13** [Mikhus/canvas-gauges#78]
Library is broken in CommonJS environment. Module is wrapped into anonymous function to provide something close to UMD module format:

    (function(ns) {'use strict';
        // code
    }(typeof module !== "undefined" ? module.exports : window));

It works perfectly in non-CommonJS environment, since it sets `ns` to `Window`. In CommonJS environment `ns` will be module exports object. But code of the library extensively uses `ns` variable to access `Window`'s properties: [1](https://github.com/Mikhus/canvas-gauges/blob/master/lib/Animation.js#L257), [2](https://github.com/Mikhus/canvas-gauges/blob/master/lib/DomObserver.js#L109), [3](https://github.com/Mikhus/canvas-gauges/blob/master/lib/DomObserver.js#L333), etc. Obviously in CommonJS environment these properties are undefined, so it fails/behaves very weird (in my case animation went insane, because `ns.performance` was undefined and it used `Date.now()` to get animation start time (and they are completely different things).

The simple solution would be just [removing `ns.` prefix](https://github.com/devoto13/canvas-gauges/commit/2333bd578f641fd67e35e11ee4897de706844f4e) from all the global variables. If you think we should go with this solution, I can prepare PR.

Check out [updated example](https://github.com/devoto13/canvas-gauges/blob/commonjs-issue-demo/examples/scripted.html) to get simple reproduction of the issue in CommonJS environment.

---

As a side note. I find the Regex based build process a bit speculative. Any reason why you use that instead something like [this babel plugin](https://babeljs.io/docs/plugins/transform-es2015-modules-umd/) for example? Maybe we can improve build process as well?

**Comment by Mikhus** [comment]
Thanks, will fix that in 2.1.0

Yes, there is a reason of such build. It provides an ability to get minimum code.

**Comment by devoto13** [comment]
Okey, thank you. Should I open PR with [my current fix](https://github.com/devoto13/canvas-gauges/commit/2333bd578f641fd67e35e11ee4897de706844f4e) or you're gonna do the work yourself?

**Comment by Mikhus** [comment]
these kind of things:

```javascript
if (matchMedia) {
 // ...
}
```

will produce uncaught ReferenceErrors in the environments where is it not defined, so such kind of fix is not acceptable.

it should be changed to something like:

```javascript
if (typeof matchMedia !== undefined) {
 // ...
}
```

or I guess it will be much better to make more generic changes over ns definition...

**Comment by devoto13** [comment]
Yes, you're right about ReferenceErrors. Probably it is also better to have more generic change for `ns`. Let me know if I can provide any help! Your library saved me hours and hours of work :)

**Comment by socmag** [comment]
sry... how about being able to specify your own time provider when creating the gauge? would that help?

**Comment by Mikhus** [comment]
Fixed in commit https://github.com/Mikhus/canvas-gauges/commit/ab1095188c6991a19e2d49cfab0c29b9824afcd1

Will be released in v2.1.0
